tests/test_special_ops.py

```diff
@@ -445,6 +445,106 @@ def test_accuracy_unique(shape, dtype, sorted, return_inverse, return_counts):
+        unique_vals = torch.randint(
+            -10, 10, (indices.max() + 1,), device="cuda", dtype=dtype
+        )
+        inp_flat = unique_vals[indices]
+        inp = inp_flat.reshape(shape).to(flag_gems.device)
+
+    ref_inp = to_reference(inp, False)
+
+    if return_counts:
+        if return_inverse:
+            with flag_gems.use_gems():
+                res_output, res_inverse_indices, res_counts = torch.unique_consecutive(
+                    inp,
+                    return_inverse=return_inverse,
+                    return_counts=return_counts,
+                    dim=dim,
+                )
+            ref_output, ref_inverse_indices, ref_counts = torch.unique_consecutive(
+                ref_inp,
+                return_inverse=return_inverse,
+                return_counts=return_counts,
+                dim=dim,
+            )
+            # assert res_output.numel() == ref_output.numel()
+            gems_assert_equal(res_inverse_indices, ref_inverse_indices)
+        else:
+            with flag_gems.use_gems():
+                res_output, res_counts = torch.unique_consecutive(
+                    inp,
+                    return_inverse=return_inverse,
+                    return_counts=return_counts,
+                    dim=dim,
+                )
+            ref_output, ref_counts = torch.unique_consecutive(
+                ref_inp,
+                return_inverse=return_inverse,
+                return_counts=return_counts,
+                dim=dim,
+            )
+            # assert res_output.numel() == ref_output.numel()
+        gems_assert_equal(res_counts, ref_counts)
+    else:
+        if return_inverse:
+            with flag_gems.use_gems():
+                res_output, res_inverse_indices = torch.unique_consecutive(
+                    inp,
+                    return_inverse=return_inverse,
+                    return_counts=return_counts,
+                    dim=dim,
+                )
+            ref_output, ref_inverse_indices = torch.unique_consecutive(
+                ref_inp,
+                return_inverse=return_inverse,
+                return_counts=return_counts,
+                dim=dim,
+            )
+            # assert res_output.numel() == ref_output.numel()
+            gems_assert_equal(res_inverse_indices, ref_inverse_indices)
+        else:
+            with flag_gems.use_gems():
+                res_output = torch.unique_consecutive(
+                    inp,
+                    return_inverse=return_inverse,
+                    return_counts=return_counts,
+                    dim=dim,
+                )
+            ref_output = torch.unique_consecutive(
+                ref_inp,
+                return_inverse=return_inverse,
+                return_counts=return_counts,
+                dim=dim,
+            )
+            # assert res_output.numel() == ref_output.numel()
+
+    gems_assert_equal(res_output, ref_output)
```

Comment:
Please consider duplicate this assertion for readability.
The current indentation is not so straightforward for maintenance.